frontend/src/app/modules/user/components/user-profile-settings/user-profile-settings.component.ts

```diff
@@ -22,24 +25,38 @@ export class UserProfileSettingsComponent extends BaseComponent implements OnIni
         private toastNotificationService: ToastNotificationService
     ) {
         super();
-        this.user = authService.getUser();
+        //this.user = authService.getUser();
     }
 
     ngOnInit(): void {
-        this.editProfileForm = new FormGroup(
-            {
-                firstName: new FormControl(this.user?.firstName),
-                lastName: new FormControl(this.user?.lastName),
-                email: new FormControl(this.user?.email)
-            }
-        );
+        this.loadUser();
+    }
+
+    submit(editForm) {
+        this.updateUser(editForm);
     }
 
-    onSubmit() {
-        this.userService.updateUser(this.user)
+    loadUser() {
+        this.userService.getUserById(this.userId).subscribe((data: User) => {
+            this.user = data;
+        });
+    }
+
+    updateUser(editForm) {
+        this.user.firstName = editForm.value.firstName;
+        this.user.lastName = editForm.value.lastName;
```

Comment:
Please use {...obj} for this. The mutation is not a good idea

Reply:
Fix